atomicapp/nulecule/container.py

```diff
@@ -0,0 +1,54 @@
+import os
+import shutil
+import subprocess
+import uuid
+import logging
+
+from atomicapp.constants import APP_ENT_PATH
+
+logger = logging.getLogger(__name__)
+
+
+class DockerHandler(object):
+    """Interface to interact with Docker."""
+
+    def __init__(self, docker_cli='/usr/bin/docker'):
+        self.docker_cli = docker_cli
+
+    def pull(self, image, update=False):
+        if not self.is_image_present(image) or update:
+            logger.debug('Pulling Docker image: %s' % image)
+            pull_cmd = [self.docker_cli, 'pull', image]
+            subprocess.call(pull_cmd)
+        else:
+            logger.debug('Skipping pulling Docker image: %s' % image)
+
+    def extract(self, image, source, dest):
+        container_id = None
+        run_cmd = [
+            self.docker_cli, 'run', '-d', '--entrypoint', '/bin/true', image]
+        logger.debug(run_cmd)
+        container_id = subprocess.check_output(run_cmd).strip()
+        tmpdir = '/tmp/nulecule-{}'.format(uuid.uuid1())
+        cp_cmd = [self.docker_cli, 'cp',
+                  '%s:/%s/.' % (container_id, source),
+                  tmpdir]
+        logger.debug('%s' % cp_cmd)
+        subprocess.call(cp_cmd)
+        shutil.copytree(os.path.join(tmpdir, APP_ENT_PATH), dest)
+        shutil.rmtree(tmpdir)
+        rm_cmd = [self.docker_cli, 'rm', '-f', container_id]
+        subprocess.call(rm_cmd)
+
+    def is_image_present(self, image):
+        output = subprocess.check_output([self.docker_cli, 'images'])
+        image_lines = output.strip().splitlines()[1:]
+        for line in image_lines:
+            words = line.split()
+            image_name = words[0]
+            registry = repo = None
+            if image_name.find('/') >= 0:
+                registry, repo = image_name.split('/', 1)
+            if image_name == image or repo == image:
+                return True
```

Comment:
Makes sense! That's actually why previous versions of Atomic App had a code to prepend the image name with registry based on `answers.conf`...though we agreed we will leave this to user's docker settings..so maybe it's time to bring it back:-) 

Problem of requiring the registry if that...

1. MyApp lives in some public registry (pblc.io) and has pblc.io/another-app as dependency
2. To lower the network load, I will pull these nulecules to my local company registry and this will have mycompany.io/MyApp and mycompany.io/another-app...
3. I do `atomic run mycompany.io/MyApp`, but it fails as MyApp still requires pblc.io/another-app